src/uu/touch/src/touch.rs

```diff
@@ -135,12 +135,72 @@ fn filetime_to_datetime(ft: &FileTime) -> Option<DateTime<Local>> {
     Some(DateTime::from_timestamp(ft.unix_seconds(), ft.nanoseconds())?.into())
 }
 
+/// Whether all characters in the string are digits.
+fn all_digits(s: &str) -> bool {
+    s.as_bytes().iter().all(u8::is_ascii_digit)
+}
+
+/// Convert a two-digit year string to the corresponding number.
+///
+/// `s` must be of length two or more. The last two bytes of `s` are
+/// assumed to be the two digits of the year.
+fn get_year(s: &str) -> u8 {
+    let bytes = s.as_bytes();
+    let n = bytes.len();
+    let y1 = bytes[n - 2] - b'0';
+    let y2 = bytes[n - 1] - b'0';
+    10 * y1 + y2
+}
+
+/// Whether the first filename should be interpreted as a timestamp.
+fn is_first_filename_timestamp(
+    reference: Option<&OsString>,
+    date: Option<&str>,
+    timestamp: &Option<String>,
+    files: &[&String],
+) -> bool {
+    match std::env::var("_POSIX2_VERSION") {
+        Ok(s) if s == "199209" => {
+            if timestamp.is_none() && reference.is_none() && date.is_none() {
+                if files.len() >= 2 {
+                    let s = files[0];
+                    if s.len() == 8 && all_digits(s) {
+                        true
+                    } else if s.len() == 10 && all_digits(s) {
+                        let year = get_year(s);
+                        (69..=99).contains(&year)
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                }
+            } else {
+                false
+            }
```

Comment:
A lot of `if`s and `else`s ;-)  What do you think about something like the following refactoring?
```suggestion
            if timestamp.is_none() && reference.is_none() && date.is_none() && files.len() >= 2 {
                let s = &files[0];
                (s.len() == 8 && all_digits(s))
                    || (s.len() == 10 && all_digits(s) && (69..=99).contains(&get_year(s)))
            } else {
                false
            }
```